### PR TITLE
projectile stuff + improved bullet interaction with station bots

### PR DESCRIPTION
**Pull request by Tarmunora**
[CHORE] [BALANCE]
## About the PR 
Cleans up/refactors some projectile stuff - mob-piercing code is less ass, meatshielding is now handled by the grab instead of the mob

Makes projectiles able to hit multiple station bots (secbots/etc) on one tile, losing some power for each bot they go through

Adds some more signal stuff to handle these things

## Why's this needed? 
Make gunning down a large stack of secbots/etc more manageable, clean up projectiles a little.

### CODE REVIEW

**Comment by Tarmunora** [comment]
wait what happens if you grab 2 people

File: _std/defines/projectiles.dm [hunk 1]
```diff
@@ -1,8 +1,16 @@
 
 //pass flags
-#define PROJ_PASSNONE 0
-#define PROJ_PASSWALL 1
-#define PROJ_PASSOBJ 2
+#define PROJ_PASSNONE			0x000
+#define PROJ_PASSWALL			0x001
+#define PROJ_PASSOBJ			0x002
```
Comment (ZeWaka) [inline _std/defines/projectiles.dm:5]: the `(1<<3)` etc. syntax is a bit less confusing than this one

File: code/modules/robotics/bot/bot_parent.dm [hunk 2]
```diff
@@ -56,12 +56,12 @@
 	bullet_act(var/obj/projectile/P)
 		if (!P || !istype(P))
 			return
-
 		hit_twitch(src)
 
 		var/damage = 0
 		damage = round(((P.power/4)*P.proj_data.ks_ratio), 1.0)
 
+		boutput(world, "took <b>[damage]</b> damage!")
```
Comment (ZeWaka) [inline code/modules/robotics/bot/bot_parent.dm:64]: .

Comment (Tarmunora) [reply code/modules/robotics/bot/bot_parent.dm:64]: fuck

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**